### PR TITLE
fix(http2_adapter):request by http2 without network once,the same domain request will never success.

### PR DESCRIPTION
**Pull request by sunhapper**
### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [x] I have updated the `CHANGELOG.md` in the corresponding package

### Additional context and info (if any)

`_ConnectionManager._connect` may throw SocketException when call `await initFuture`,if it occurs, even if the network is restored, subsequent requests with the same domain will still obtain the cached initFuture, and it will still fail

### CODE REVIEW

File: plugins/http2_adapter/lib/src/connection_manager_imp.dart [hunk 1]
```diff
@@ -45,12 +45,13 @@ class _ConnectionManager implements ConnectionManager {
       if (initFuture == null) {
         _connectFutures[domain] = initFuture = _connect(options);
       }
-      transportState = await initFuture;
+      transportState = await initFuture.whenComplete(() {
+         _connectFutures.remove(domain);
```
Comment (AlexV525) [inline plugins/http2_adapter/lib/src/connection_manager_imp.dart:49]: Can we remove this only when the future failed?

Comment (sunhapper) [reply plugins/http2_adapter/lib/src/connection_manager_imp.dart]: I changed code only removed cached Future on fail and restored the logic of success to its original state.

**Review by AlexV525 — APPROVED** [review]

File: plugins/http2_adapter/lib/src/connection_manager_imp.dart [hunk 2]
```diff
@@ -45,7 +45,12 @@ class _ConnectionManager implements ConnectionManager {
       if (initFuture == null) {
         _connectFutures[domain] = initFuture = _connect(options);
       }
-      transportState = await initFuture;
+      try {
+        transportState = await initFuture;
+      } catch (e) {
```
Comment (AlexV525) [inline plugins/http2_adapter/lib/src/connection_manager_imp.dart:50]:
```suggestion
      } catch (_) {
```

Comment (sunhapper) [reply plugins/http2_adapter/lib/src/connection_manager_imp.dart]: Thanks  a lot for the update

**Pull request merged**